test/e2e/alertmanager_instance_namespaces_test.go

```diff
@@ -26,8 +27,9 @@ import (
 )
 
 func testAlertmanagerInstanceNamespacesAllNs(t *testing.T) {
-	ctx := framework.NewTestCtx(t)
-	defer ctx.Cleanup(t)
+	ctx := context.Background()
+	testCtx := framework.NewTestCtx(t)
```

Comment:
Since `testCtx` is oftentimes passed down to functions along with `ctx`, I think we should put the context inside of the TestCtx structure. This is a valid use case for putting context inside of a structure since the structure itself is meant only to define the lifecycle of the test so its goal is clear.

Reply:
I remember we had added ctx like that previously inside testCtx. But since it's not recommended approach should we avoid that?

Reply:
Yes, it is not recommended to add a context to structs since it would lead to some confusion but in our case, we are just adding it to a structure that defines a context, so I think it is a valid use case.

Also, maybe a better approach would be to update the TextCtx structure to extend the context.Context interface with the [testCtx Cleanup](https://github.com/prometheus-operator/prometheus-operator/blob/a10ba4a4e57144a21c9be6c4e5fad456b0433518/test/framework/context.go#L59-L69) function as a cancel function.

Reply:
Sorry for the late review. I think it would even be clearer to instantiate a new `context.Background()` for each function call that expect a context (e.g no need to pass a unique context from top-level to down). 

Instead of 
https://github.com/prometheus-operator/prometheus-operator/blob/83fe36566f4e0894eb5ffcd2638a0f039a17bdeb/test/framework/alertmanager.go#L124

We can do

```golang
 _, err = f.KubeClient.CoreV1().Secrets(ns).Create(context.Background(), s, metav1.CreateOptions{})
```

From https://pkg.go.dev/context#Background:

> Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and **tests**, and as the top-level Context for incoming requests.

Reply:
@simonpasquier  I have passed ctx variable which is declared in each test 

like in https://github.com/prometheus-operator/prometheus-operator/pull/4183/files#diff-06a2e1bc05a7f1185bff95c6029e8b10a257bee7b89ffa57b226ca2d19f24498R47

Do you think current change is ok?

Reply:
My current idea would be to avoid passing a top-level context around because in the end, it's always `context.Background()` anyway. 
In practice, the change would be to remove the `Ctx` field from the `framework.Framework` struct and everywhere we used to have `f.Ctx`, replace by `context.Background()`.

Reply:
Since current update is creating a variable to hold context.Background() like [here](https://github.com/prometheus-operator/prometheus-operator/pull/4183/files#diff-bd371abb25ee1c95090cf7508fc3133e1fb6faae26c4c01fdac53dd759052d3eR31). I felt that looks good to read than creating a new context object in every function call needing a context to be passed.

Reply:
After discussing offline we decided to use context.Background places where both ctx and testCtx is passed for better readability of code